lib/concurrent/edge/future.rb

```diff
@@ -511,39 +583,64 @@ def pr_async_callback_on_completion(success, value, reason, executor, callback)
       end
     end
 
+    # As Future but applies its Array value correctly to blocks.
+    class ArrayFuture < Future
+      # @!visibility private
+      def apply_value(value, block)
+        block.call(*value)
+      end
+    end
+
+    # A Event which can be completed by user.
     class CompletableEvent < Event
-      # Complete the event
+      # Complete the Event, `raise` if already completed
       def complete(raise = true)
```

Comment:
Since `raise` is already a defined method in Ruby, it may be wise to rename this arg.